pkg/stub/handler.go

```diff
@@ -2,96 +2,436 @@ package stub
+
+	fileimagegetter    ImageStreamFromFileGetter
+	filetemplategetter TemplateFromFileGetter
+	filefinder         ResourceFileLister
+
+	namespace string
+
+	skippedTemplates    map[string]bool
+	skippedImagestreams map[string]bool
 }
 
-func (h *Handler) Handle(ctx context.Context, event sdk.Event) error {
-	switch o := event.Object.(type) {
-	case *v1alpha1.SamplesResource:
-		logrus.Printf("ctrl cfg %#v", o)
-		restconfig, err := getRestConfig()
+func (h *Handler) manageDockerCfgSecret(deleted bool, operatorCfg *v1alpha1.SamplesResource, secret *corev1.Secret) error {
+	// clear out temporary secret cache
+	h.registrySecret = corev1.Secret{}
+	var newStatus corev1.ConditionStatus
+	if deleted {
+		err := h.secretclientwrapper.Delete("openshift", secret.Name, &metav1.DeleteOptions{})
+		if err != nil && !kerrors.IsNotFound(err) {
+			return h.processError(operatorCfg, v1alpha1.SecretUpdateFailed, err, "failed to delete before create dockerconfig secret the openshift namespace: %v")
+		}
+		logrus.Printf("registry dockerconfig secret %s was deleted", secret.Name)
+		newStatus = corev1.ConditionFalse
+	} else {
+		secretToCreate := corev1.Secret{}
+		secret.DeepCopyInto(&secretToCreate)
+		secretToCreate.Namespace = ""
+		secretToCreate.ResourceVersion = ""
+
+		s, err := h.secretclientwrapper.Get("openshift", secretToCreate.Name)
+		if err != nil && !kerrors.IsNotFound(err) {
+			return h.processError(operatorCfg, v1alpha1.SecretUpdateFailed, err, "failed to get registry dockerconfig secret in openshift namespace : %v")
+		}
 		if err != nil {
-			logrus.Errorf("failed to get rest config : %v", err)
-			return err
+			s = nil
+		}
+		if s != nil {
+			_, err = h.secretclientwrapper.Update("openshift", &secretToCreate)
+		} else {
+			_, err = h.secretclientwrapper.Create("openshift", &secretToCreate)
 		}
-		tempclient, err := getTemplateClient(restconfig)
 		if err != nil {
-			logrus.Errorf("failed to get template client : %v", err)
-			return err
+			return h.processError(operatorCfg, v1alpha1.SecretUpdateFailed, err, "failed to create/update registry dockerconfig secret in openshif namespace : %v")
+		}
+		logrus.Printf("created/updated dockerconfig secret %s in openshift namespace", operatorCfg.Spec.RegistryCredentials.Name)
+		newStatus = corev1.ConditionTrue
+	}
+
+	now := kapis.Now()
+	status := operatorCfg.Condition(v1alpha1.ImportCredentialsExists)
+	status.LastUpdateTime = now
+	if status.Status != newStatus {
+		status.Status = newStatus
+		status.LastTransitionTime = now
+	}
+	operatorCfg.ConditionUpdate(status)
+	err := h.sdkwrapper.Update(operatorCfg)
+	if err != nil {
+		if !kerrors.IsConflict(err) {
+			return h.processError(operatorCfg, v1alpha1.SamplesUpdateFailed, err, "failed adding success condition to config: %v")
+		}
+		logrus.Println("got conflict error on success status update, just going to forgo this time vs. retry")
+		// just return the raw error and initiate the sdk's requeue/ratelimiter setup
+		return err
+	}
```

Comment:
In the event of duplicate events, the Last Update Time would be more accurate if we did the updating regardless of the status equality.  It would correspond to the last event to come in.

Depends on how close together the duplicate events were.

Hence I did it this way.

Though I don't feel strongly about it if you are not concerned about updating the last access time in this case.... just let me know.

Reply:
saw comment below re: spamming master ... changing